--- kexi/migration/xbase/xbasemigrate.cpp.orig	2013-02-14 09:17:23.592745008 +0000
+++ kexi/migration/xbase/xbasemigrate.cpp	2013-02-14 09:17:35.050744333 +0000
@@ -253,18 +253,11 @@
           blobFieldLength = tableDbf->GetMemoFieldLen(j);
           memoBuffer = new char[blobFieldLength];
 
-          #ifdef XB_LOCKING_ON
-            tableDbf->LockMemoFile( F_SETLK, F_RDLCK );
-          #endif
-
           if ( ( returnCode = tableDbf->GetMemoField( j , blobFieldLength, memoBuffer, F_SETLKW ) ) != XB_NO_ERROR ) {
             kDebug()<<"Error reading blob field. Error code: "<<returnCode; // make error message more verbose
           } else {
             val = KexiDB::cstringToVariant( memoBuffer, fieldsExpanded.at(j)->field, blobFieldLength );
           }
-          #ifdef XB_LOCKING_ON
-            tableDbf->LockMemoFile( F_SETLK, F_UNLCK );
-          #endif
 
           break;
         #else
