--- calligra-2.8.0/CMakeLists.txt.staging~	2014-03-06 13:38:27.059248823 +0100
+++ calligra-2.8.0/CMakeLists.txt	2014-03-06 13:40:12.945861359 +0100
@@ -74,9 +74,9 @@ endif(NOT DEFINED RELEASE_BUILD)
 message(STATUS "Release build: ${RELEASE_BUILD}")
 
 if (RELEASE_BUILD)
-    set(CALLIGRA_SHOULD_BUILD_STAGING FALSE)
+	option(CALLIGRA_SHOULD_BUILD_STAGING "Build staging bits" OFF)
 else (RELEASE_BUILD)
-    set(CALLIGRA_SHOULD_BUILD_STAGING TRUE)
+	option(CALLIGRA_SHOULD_BUILD_STAGING "Build staging bits" ON)
 endif (RELEASE_BUILD)
 
 if (WIN32)
