--- calligra-2.9.1/CMakeLists.txt.staging~	2015-03-17 00:31:07.201603139 +0100
+++ calligra-2.9.1/CMakeLists.txt	2015-03-17 00:32:01.278769370 +0100
@@ -129,9 +129,9 @@ if(NOT PRODUCTSET)
 endif()
 
 if (RELEASE_BUILD)
-    set(CALLIGRA_SHOULD_BUILD_STAGING FALSE)
+    option(CALLIGRA_SHOULD_BUILD_STAGING "Build staging bits" FALSE)
 else ()
-    set(CALLIGRA_SHOULD_BUILD_STAGING TRUE)
+    option(CALLIGRA_SHOULD_BUILD_STAGING "Build staging bits" TRUE)
 endif ()
 
 # finally choose products/features to build
