diff -up calligra-25.08.2/filters/karbon/pdf/SvgOutputDev.cpp.omv~ calligra-25.08.2/filters/karbon/pdf/SvgOutputDev.cpp
--- calligra-25.08.2/filters/karbon/pdf/SvgOutputDev.cpp.omv~	2025-11-04 20:31:42.628195298 +0100
+++ calligra-25.08.2/filters/karbon/pdf/SvgOutputDev.cpp	2025-11-04 20:31:56.297098824 +0100
@@ -390,7 +390,7 @@ void SvgOutputDev::drawString(GfxState *
         return;
 
     // ignore empty strings
-    if (s->getLength() == 0)
+    if (s->size() == 0)
         return;
 
 #if POPPLER_VERSION_MACRO < QT_VERSION_CHECK(22, 04, 0)
@@ -402,7 +402,7 @@ void SvgOutputDev::drawString(GfxState *
     QString str;
 
     const char *p = s->c_str();
-    int len = s->getLength();
+    int len = s->size();
     CharCode code;
     const Unicode *u = nullptr;
     int uLen;
