--- calligra-2.9.0/CMakeLists.txt.staging~	2015-02-27 15:46:47.818311149 +0100
+++ calligra-2.9.0/CMakeLists.txt	2015-02-27 15:48:08.800943384 +0100
@@ -129,9 +129,9 @@ if(NOT PRODUCTSET)
 endif(NOT PRODUCTSET)
 
 if (RELEASE_BUILD)
-    set(CALLIGRA_SHOULD_BUILD_STAGING FALSE)
+    option(CALLIGRA_SHOULD_BUILD_STAGING "Build staging bits" OFF)
 else (RELEASE_BUILD)
-    set(CALLIGRA_SHOULD_BUILD_STAGING TRUE)
+    option(CALLIGRA_SHOULD_BUILD_STAGING "Build staging bits" ON)
 endif (RELEASE_BUILD)
 
 # finally choose products/features to build
