diff -up calligra-master/plugins/spacenavigator/SpaceNavigatorEvent.cpp.omv~ calligra-master/plugins/spacenavigator/SpaceNavigatorEvent.cpp
--- calligra-master/plugins/spacenavigator/SpaceNavigatorEvent.cpp.omv~	2024-08-06 12:48:02.659091105 +0200
+++ calligra-master/plugins/spacenavigator/SpaceNavigatorEvent.cpp	2024-08-06 12:48:07.619130242 +0200
@@ -8,7 +8,7 @@
 #include <KoPointerEvent.h>
 
 SpaceNavigatorEvent::SpaceNavigatorEvent(KoInputDeviceHandlerEvent::Type type)
-    : KoInputDeviceHandlerEvent(type)
+    : KoInputDeviceHandlerEvent(type, nullptr)
 {
 }
 
