diff -up calligra-3.2.1/libs/text/CMakeLists.txt.omv~ calligra-3.2.1/libs/text/CMakeLists.txt
--- calligra-3.2.1/libs/text/CMakeLists.txt.omv~	2021-12-09 22:30:19.371282836 +0100
+++ calligra-3.2.1/libs/text/CMakeLists.txt	2021-12-09 22:30:58.958598838 +0100
@@ -152,7 +152,7 @@ if( SHOULD_BUILD_FEATURE_RDF )
 endif()
 
 if( FONTCONFIG_FOUND )
-    target_link_libraries(kotext PRIVATE ${FONTCONFIG_LIBRARIES})
+    target_link_libraries(kotext PRIVATE ${FONTCONFIG_LIBRARIES} -lfontconfig)
 endif()
 
 if( FREETYPE_FOUND )
