diff -up calligra-3.1.0/filters/karbon/pdf/SvgOutputDev.cpp.omv~ calligra-3.1.0/filters/karbon/pdf/SvgOutputDev.cpp
--- calligra-3.1.0/filters/karbon/pdf/SvgOutputDev.cpp.omv~	2019-02-03 18:52:27.325779974 +0100
+++ calligra-3.1.0/filters/karbon/pdf/SvgOutputDev.cpp	2019-02-03 18:53:25.408436102 +0100
@@ -402,7 +402,7 @@ void SvgOutputDev::drawString(GfxState *
 
     QString str;
 
-    const char * p = s->getCString();
+    const char * p = s->c_str();
     int len = s->getLength();
     CharCode code;
     Unicode *u = nullptr;
@@ -451,11 +451,11 @@ void SvgOutputDev::drawString(GfxState *
     *d->body << " y=\"" << y << "px\"";
 
     if (font && font->getFamily()) {
-        *d->body << " font-family=\"" << QString::fromLatin1(font->getFamily()->getCString()) << "\"";
-        //debugPdf << "font family:" << QString::fromLatin1( font->getFamily()->getCString() );
+        *d->body << " font-family=\"" << QString::fromLatin1(font->getFamily()->c_str()) << "\"";
+        //debugPdf << "font family:" << QString::fromLatin1( font->getFamily()->c_str() );
     } else if (font && font->getName()) {
-        *d->body << " font-family=\"" << QString::fromLatin1(font->getName()->getCString()) << "\"";
-        //debugPdf << "font name:" << QString::fromLatin1( font->getName()->getCString() );
+        *d->body << " font-family=\"" << QString::fromLatin1(font->getName()->c_str()) << "\"";
+        //debugPdf << "font name:" << QString::fromLatin1( font->getName()->c_str() );
     }
     *d->body << " font-size=\"" << qMax(state->getFontSize(), state->getTransformedFontSize()) << "px\"";
 
