diff -up calligra-3.1.90/filters/karbon/wmf/WmfExport.cpp.omv~ calligra-3.1.90/filters/karbon/wmf/WmfExport.cpp
--- calligra-3.1.90/filters/karbon/wmf/WmfExport.cpp.omv~	2020-04-08 18:42:52.307832736 +0200
+++ calligra-3.1.90/filters/karbon/wmf/WmfExport.cpp	2020-04-08 18:42:59.281045240 +0200
@@ -35,6 +35,7 @@
 #include <KoUnit.h>
 #include <KoPAPageBase.h>
 #include <KoPageLayout.h>
+#include <QPainterPath>
 
 #include <algorithm>
 
diff -up calligra-3.1.90/filters/libmso/shapes.cpp.omv~ calligra-3.1.90/filters/libmso/shapes.cpp
--- calligra-3.1.90/filters/libmso/shapes.cpp.omv~	2020-04-08 16:20:40.416514140 +0200
+++ calligra-3.1.90/filters/libmso/shapes.cpp	2020-04-08 16:20:47.749308627 +0200
@@ -38,6 +38,7 @@
 #include <QTransform>
 #include <QBuffer>
 #include <QUrl>
+#include <QPainterPath>
 
 #include <cmath>
 
diff -up calligra-3.1.90/karbon/common/KarbonOutlinePaintingStrategy.cpp.omv~ calligra-3.1.90/karbon/common/KarbonOutlinePaintingStrategy.cpp
--- calligra-3.1.90/karbon/common/KarbonOutlinePaintingStrategy.cpp.omv~	2020-04-08 17:50:42.020053321 +0200
+++ calligra-3.1.90/karbon/common/KarbonOutlinePaintingStrategy.cpp	2020-04-08 17:50:47.445036919 +0200
@@ -25,6 +25,7 @@
 
 #include <QPen>
 #include <QPainter>
+#include <QPainterPath>
 
 class OutlineStroke : public KoShapeStroke {
 public:
diff -up calligra-3.1.90/karbon/ui/widgets/KarbonSmallStylePreview.cpp.omv~ calligra-3.1.90/karbon/ui/widgets/KarbonSmallStylePreview.cpp
--- calligra-3.1.90/karbon/ui/widgets/KarbonSmallStylePreview.cpp.omv~	2020-04-08 17:51:34.432973678 +0200
+++ calligra-3.1.90/karbon/ui/widgets/KarbonSmallStylePreview.cpp	2020-04-08 17:51:38.812974727 +0200
@@ -40,6 +40,7 @@
 #include <QPushButton>
 #include <QLabel>
 #include <QPainter>
+#include <QPainterPath>
 #include <QPaintEvent>
 #include <QHBoxLayout>
 
diff -up calligra-3.1.90/libs/basicflakes/tools/KoCreatePathTool_p.h.omv~ calligra-3.1.90/libs/basicflakes/tools/KoCreatePathTool_p.h
--- calligra-3.1.90/libs/basicflakes/tools/KoCreatePathTool_p.h.omv~	2020-04-08 17:56:53.353514834 +0200
+++ calligra-3.1.90/libs/basicflakes/tools/KoCreatePathTool_p.h	2020-04-08 17:57:02.142642247 +0200
@@ -34,6 +34,8 @@
 #include "KoSnapStrategy.h"
 #include "KoToolBase_p.h"
 
+#include <QPainterPath>
+
 #include "math.h"
 
 /// Small helper to keep track of a path point and its parent path shape
diff -up calligra-3.1.90/libs/flake/KoConnectionShape.cpp.omv~ calligra-3.1.90/libs/flake/KoConnectionShape.cpp
--- calligra-3.1.90/libs/flake/KoConnectionShape.cpp.omv~	2020-04-08 16:20:16.900138930 +0200
+++ calligra-3.1.90/libs/flake/KoConnectionShape.cpp	2020-04-08 16:20:24.962930971 +0200
@@ -34,6 +34,7 @@
 #include <KoXmlNS.h>
 #include <KoUnit.h>
 #include <QPainter>
+#include <QPainterPath>
 
 #include <FlakeDebug.h>
 
diff -up calligra-3.1.90/libs/flake/KoHatchBackground.cpp.omv~ calligra-3.1.90/libs/flake/KoHatchBackground.cpp
--- calligra-3.1.90/libs/flake/KoHatchBackground.cpp.omv~	2020-04-08 16:22:36.431840461 +0200
+++ calligra-3.1.90/libs/flake/KoHatchBackground.cpp	2020-04-08 16:22:40.699692474 +0200
@@ -34,6 +34,7 @@
 #include <QColor>
 #include <QString>
 #include <QPainter>
+#include <QPainterPath>
 
 class KoHatchBackgroundPrivate : public KoColorBackgroundPrivate
 {
diff -up calligra-3.1.90/libs/flake/KoOdfGradientBackground.cpp.omv~ calligra-3.1.90/libs/flake/KoOdfGradientBackground.cpp
--- calligra-3.1.90/libs/flake/KoOdfGradientBackground.cpp.omv~	2020-04-08 16:21:51.734345842 +0200
+++ calligra-3.1.90/libs/flake/KoOdfGradientBackground.cpp	2020-04-08 16:21:56.640185066 +0200
@@ -31,6 +31,7 @@
 #include <KoStyleStack.h>
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QColor>
 #include <QImage>
 #include <qmath.h>
diff -up calligra-3.1.90/libs/flake/KoPathShape.cpp.omv~ calligra-3.1.90/libs/flake/KoPathShape.cpp
--- calligra-3.1.90/libs/flake/KoPathShape.cpp.omv~	2020-04-08 16:17:47.441058911 +0200
+++ calligra-3.1.90/libs/flake/KoPathShape.cpp	2020-04-08 16:17:54.406089415 +0200
@@ -50,6 +50,7 @@
 
 #include <FlakeDebug.h>
 #include <QPainter>
+#include <QPainterPath>
 
 #include <qnumeric.h> // for qIsNaN
 static bool qIsNaNPoint(const QPointF &p) {
diff -up calligra-3.1.90/libs/flake/KoPatternBackground.cpp.omv~ calligra-3.1.90/libs/flake/KoPatternBackground.cpp
--- calligra-3.1.90/libs/flake/KoPatternBackground.cpp.omv~	2020-04-08 16:18:43.700890104 +0200
+++ calligra-3.1.90/libs/flake/KoPatternBackground.cpp	2020-04-08 16:18:51.695799788 +0200
@@ -36,6 +36,7 @@
 
 #include <QBrush>
 #include <QPainter>
+#include <QPainterPath>
 
 class KoPatternBackgroundPrivate : public KoShapeBackgroundPrivate
 {
diff -up calligra-3.1.90/libs/flake/KoShapeManager.cpp.omv~ calligra-3.1.90/libs/flake/KoShapeManager.cpp
--- calligra-3.1.90/libs/flake/KoShapeManager.cpp.omv~	2020-04-08 16:19:52.967712707 +0200
+++ calligra-3.1.90/libs/flake/KoShapeManager.cpp	2020-04-08 16:20:05.027432144 +0200
@@ -44,6 +44,7 @@
 #include "KoShapePaintingContext.h"
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QTimer>
 #include <FlakeDebug.h>
 
diff -up calligra-3.1.90/libs/flake/KoShapeShadow.cpp.omv~ calligra-3.1.90/libs/flake/KoShapeShadow.cpp
--- calligra-3.1.90/libs/flake/KoShapeShadow.cpp.omv~	2020-04-08 16:34:20.614571997 +0200
+++ calligra-3.1.90/libs/flake/KoShapeShadow.cpp	2020-04-08 16:34:23.869515825 +0200
@@ -32,6 +32,7 @@
 #include <KoViewConverter.h>
 #include <FlakeDebug.h>
 #include <QPainter>
+#include <QPainterPath>
 #include <QAtomicInt>
 #include <QImage>
 #include <QRectF>
diff -up calligra-3.1.90/libs/flake/KoSnapGuide.cpp.omv~ calligra-3.1.90/libs/flake/KoSnapGuide.cpp
--- calligra-3.1.90/libs/flake/KoSnapGuide.cpp.omv~	2020-04-08 16:32:35.041165309 +0200
+++ calligra-3.1.90/libs/flake/KoSnapGuide.cpp	2020-04-08 16:32:39.912102969 +0200
@@ -27,6 +27,7 @@
 #include <KoCanvasBase.h>
 
 #include <QPainter>
+#include <QPainterPath>
 
 #include <math.h>
 
diff -up calligra-3.1.90/libs/flake/KoSnapStrategy.cpp.omv~ calligra-3.1.90/libs/flake/KoSnapStrategy.cpp
--- calligra-3.1.90/libs/flake/KoSnapStrategy.cpp.omv~	2020-04-08 16:18:24.841044180 +0200
+++ calligra-3.1.90/libs/flake/KoSnapStrategy.cpp	2020-04-08 16:18:30.612006311 +0200
@@ -28,6 +28,7 @@
 #include <KoGuidesData.h>
 
 #include <QPainter>
+#include <QPainterPath>
 
 #include <cmath>
 
diff -up calligra-3.1.90/libs/flake/KoUnavailShape.cpp.omv~ calligra-3.1.90/libs/flake/KoUnavailShape.cpp
--- calligra-3.1.90/libs/flake/KoUnavailShape.cpp.omv~	2020-04-08 16:21:09.439674768 +0200
+++ calligra-3.1.90/libs/flake/KoUnavailShape.cpp	2020-04-08 16:21:17.883418513 +0200
@@ -25,6 +25,7 @@
 // Qt
 #include <QPen>
 #include <QPainter>
+#include <QPainterPath>
 #include <QByteArray>
 #include <QBuffer>
 #include <QDataStream>
diff -up calligra-3.1.90/libs/flake/svg/SvgParser.cpp.omv~ calligra-3.1.90/libs/flake/svg/SvgParser.cpp
--- calligra-3.1.90/libs/flake/svg/SvgParser.cpp.omv~	2020-04-08 16:22:13.450625335 +0200
+++ calligra-3.1.90/libs/flake/svg/SvgParser.cpp	2020-04-08 16:22:22.281326254 +0200
@@ -50,6 +50,7 @@
 #include <FlakeDebug.h>
 
 #include <QColor>
+#include <QPainterPath>
 
 
 SvgParser::SvgParser(KoDocumentResourceManager *documentResourceManager)
diff -up calligra-3.1.90/libs/flake/svg/SvgStyleWriter.cpp.omv~ calligra-3.1.90/libs/flake/svg/SvgStyleWriter.cpp
--- calligra-3.1.90/libs/flake/svg/SvgStyleWriter.cpp.omv~	2020-04-08 16:34:41.016214037 +0200
+++ calligra-3.1.90/libs/flake/svg/SvgStyleWriter.cpp	2020-04-08 16:34:50.944034964 +0200
@@ -54,6 +54,7 @@
 #include <QRadialGradient>
 #include <QMimeDatabase>
 #include <QMimeType>
+#include <QPainterPath>
 
 void SvgStyleWriter::saveSvgStyle(KoShape *shape, SvgSavingContext &context)
 {
diff -up calligra-3.1.90/libs/flake/tests/TestShapeBackgroundCommand.cpp.omv~ calligra-3.1.90/libs/flake/tests/TestShapeBackgroundCommand.cpp
--- calligra-3.1.90/libs/flake/tests/TestShapeBackgroundCommand.cpp.omv~	2020-04-08 17:54:47.657986760 +0200
+++ calligra-3.1.90/libs/flake/tests/TestShapeBackgroundCommand.cpp	2020-04-08 17:54:54.450054065 +0200
@@ -25,6 +25,7 @@
 #include "KoViewConverter.h"
 
 #include <QTest>
+#include <QPainterPath>
 
 void TestShapeBackgroundCommand::refCounting()
 {
diff -up calligra-3.1.90/libs/flake/tests/TestSnapStrategy.cpp.omv~ calligra-3.1.90/libs/flake/tests/TestSnapStrategy.cpp
--- calligra-3.1.90/libs/flake/tests/TestSnapStrategy.cpp.omv~	2020-04-08 17:55:20.159325582 +0200
+++ calligra-3.1.90/libs/flake/tests/TestSnapStrategy.cpp	2020-04-08 17:55:29.180426973 +0200
@@ -19,6 +19,7 @@
 
 #include "TestSnapStrategy.h"
 #include <QTest>
+#include <QPainterPath>
 #include "KoSnapStrategy.h"
 #include "KoPathShape.h"
 #include "KoSnapProxy.h"
diff -up calligra-3.1.90/libs/flake/tools/KoPathTool.cpp.omv~ calligra-3.1.90/libs/flake/tools/KoPathTool.cpp
--- calligra-3.1.90/libs/flake/tools/KoPathTool.cpp.omv~	2020-04-08 16:28:55.573230324 +0200
+++ calligra-3.1.90/libs/flake/tools/KoPathTool.cpp	2020-04-08 16:28:59.664251964 +0200
@@ -56,6 +56,7 @@
 #include <FlakeDebug.h>
 #include <klocalizedstring.h>
 #include <QPainter>
+#include <QPainterPath>
 #include <QBitmap>
 #include <QTabWidget>
 
diff -up calligra-3.1.90/libs/pageapp/KoPADocumentModel.cpp.omv~ calligra-3.1.90/libs/pageapp/KoPADocumentModel.cpp
--- calligra-3.1.90/libs/pageapp/KoPADocumentModel.cpp.omv~	2020-04-08 16:52:15.456439199 +0200
+++ calligra-3.1.90/libs/pageapp/KoPADocumentModel.cpp	2020-04-08 16:52:27.410485926 +0200
@@ -48,6 +48,7 @@
 #include <QApplication>
 #include <QClipboard>
 #include <QMenu>
+#include <QPainterPath>
 
 #include "commands/KoPAPageMoveCommand.h"
 
diff -up calligra-3.1.90/libs/pageapp/KoPAPageBase.cpp.omv~ calligra-3.1.90/libs/pageapp/KoPAPageBase.cpp
--- calligra-3.1.90/libs/pageapp/KoPAPageBase.cpp.omv~	2020-04-08 16:51:28.580199513 +0200
+++ calligra-3.1.90/libs/pageapp/KoPAPageBase.cpp	2020-04-08 16:51:36.216244899 +0200
@@ -40,6 +40,7 @@
 #include <PageAppDebug.h>
 
 #include <QPainter>
+#include <QPainterPath>
 
 #include <algorithm>
 
diff -up calligra-3.1.90/libs/vectorimage/libemf/EmfOutputPainterStrategy.cpp.omv~ calligra-3.1.90/libs/vectorimage/libemf/EmfOutputPainterStrategy.cpp
--- calligra-3.1.90/libs/vectorimage/libemf/EmfOutputPainterStrategy.cpp.omv~	2020-04-08 16:19:11.538518707 +0200
+++ calligra-3.1.90/libs/vectorimage/libemf/EmfOutputPainterStrategy.cpp	2020-04-08 16:19:29.644199382 +0200
@@ -22,6 +22,8 @@
 
 #include <VectorImageDebug.h>
 
+#include <QPainterPath>
+
 #include "EmfObjects.h"
 
 
diff -up calligra-3.1.90/libs/vectorimage/libsvm/SvmPainterBackend.cpp.omv~ calligra-3.1.90/libs/vectorimage/libsvm/SvmPainterBackend.cpp
--- calligra-3.1.90/libs/vectorimage/libsvm/SvmPainterBackend.cpp.omv~	2020-04-08 17:59:09.715746780 +0200
+++ calligra-3.1.90/libs/vectorimage/libsvm/SvmPainterBackend.cpp	2020-04-08 17:59:15.923860290 +0200
@@ -25,6 +25,7 @@
 #include <QPolygon>
 #include <QString>
 #include <QPainter>
+#include <QPainterPath>
 
 // Libsvm
 #include "SvmEnums.h"
diff -up calligra-3.1.90/libs/widgets/KoMarkerItemDelegate.cpp.omv~ calligra-3.1.90/libs/widgets/KoMarkerItemDelegate.cpp
--- calligra-3.1.90/libs/widgets/KoMarkerItemDelegate.cpp.omv~	2020-04-08 16:42:27.866479627 +0200
+++ calligra-3.1.90/libs/widgets/KoMarkerItemDelegate.cpp	2020-04-08 16:42:32.780715084 +0200
@@ -24,6 +24,7 @@
 #include <KoMarker.h>
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QPen>
 
 KoMarkerItemDelegate::KoMarkerItemDelegate(KoMarkerData::MarkerPosition position, QObject *parent)
diff -up calligra-3.1.90/plugins/chartshape/ChartShape.cpp.omv~ calligra-3.1.90/plugins/chartshape/ChartShape.cpp
--- calligra-3.1.90/plugins/chartshape/ChartShape.cpp.omv~	2020-04-08 18:11:16.876263147 +0200
+++ calligra-3.1.90/plugins/chartshape/ChartShape.cpp	2020-04-08 18:11:21.783399514 +0200
@@ -31,6 +31,7 @@
 // Qt
 #include <QPointF>
 #include <QPainter>
+#include <QPainterPath>
 #include <QSizeF>
 #include <QTextDocument>
 #include <QUrl>
diff -up calligra-3.1.90/plugins/chartshape/Legend.cpp.omv~ calligra-3.1.90/plugins/chartshape/Legend.cpp
--- calligra-3.1.90/plugins/chartshape/Legend.cpp.omv~	2020-04-08 18:08:22.875531464 +0200
+++ calligra-3.1.90/plugins/chartshape/Legend.cpp	2020-04-08 18:08:29.871717361 +0200
@@ -30,6 +30,7 @@
 #include <QBrush>
 #include <QFont>
 #include <QImage>
+#include <QPainterPath>
 
 // Calligra
 #include <KoXmlReader.h>
diff -up calligra-3.1.90/plugins/chartshape/PlotArea.cpp.omv~ calligra-3.1.90/plugins/chartshape/PlotArea.cpp
--- calligra-3.1.90/plugins/chartshape/PlotArea.cpp.omv~	2020-04-08 18:09:24.868192313 +0200
+++ calligra-3.1.90/plugins/chartshape/PlotArea.cpp	2020-04-08 18:09:31.954384038 +0200
@@ -29,6 +29,7 @@
 #include <QList>
 #include <QImage>
 #include <QPainter>
+#include <QPainterPath>
 
 // Calligra
 #include <KoXmlNS.h>
diff -up calligra-3.1.90/plugins/defaultTools/defaulttool/DefaultTool.cpp.omv~ calligra-3.1.90/plugins/defaultTools/defaulttool/DefaultTool.cpp
--- calligra-3.1.90/plugins/defaultTools/defaulttool/DefaultTool.cpp.omv~	2020-04-08 18:03:48.788635106 +0200
+++ calligra-3.1.90/plugins/defaultTools/defaulttool/DefaultTool.cpp	2020-04-08 18:03:58.870872871 +0200
@@ -65,6 +65,7 @@
 #include <QKeyEvent>
 #include <QClipboard>
 #include <QStandardPaths>
+#include <QPainterPath>
 
 #include <math.h>
 
diff -up calligra-3.1.90/plugins/formulashape/elements/BasicElement.cpp.omv~ calligra-3.1.90/plugins/formulashape/elements/BasicElement.cpp
--- calligra-3.1.90/plugins/formulashape/elements/BasicElement.cpp.omv~	2020-04-08 18:38:29.374080954 +0200
+++ calligra-3.1.90/plugins/formulashape/elements/BasicElement.cpp	2020-04-08 18:38:35.540270216 +0200
@@ -32,6 +32,7 @@
 #include <KoXmlReader.h>
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QVariant>
 
 
diff -up calligra-3.1.90/plugins/formulashape/elements/FixedElement.cpp.omv~ calligra-3.1.90/plugins/formulashape/elements/FixedElement.cpp
--- calligra-3.1.90/plugins/formulashape/elements/FixedElement.cpp.omv~	2020-04-08 18:39:19.599622820 +0200
+++ calligra-3.1.90/plugins/formulashape/elements/FixedElement.cpp	2020-04-08 18:39:24.800782523 +0200
@@ -25,6 +25,7 @@
 #include <KoXmlReader.h>
 
 #include <QPainter>
+#include <QPainterPath>
 
 
 FixedElement::FixedElement( BasicElement* parent ) : BasicElement( parent )
diff -up calligra-3.1.90/plugins/karbonplugins/tools/CalligraphyTool/KarbonCalligraphicShape.cpp.omv~ calligra-3.1.90/plugins/karbonplugins/tools/CalligraphyTool/KarbonCalligraphicShape.cpp
--- calligra-3.1.90/plugins/karbonplugins/tools/CalligraphyTool/KarbonCalligraphicShape.cpp.omv~	2020-04-08 18:42:12.269797120 +0200
+++ calligra-3.1.90/plugins/karbonplugins/tools/CalligraphyTool/KarbonCalligraphicShape.cpp	2020-04-08 18:42:18.806932254 +0200
@@ -27,6 +27,7 @@
 
 #include <QDebug>
 #include <QColor>
+#include <QPainterPath>
 
 #include <cmath>
 #include <cstdlib>
diff -up calligra-3.1.90/plugins/karbonplugins/tools/CalligraphyTool/KarbonCalligraphyTool.h.omv~ calligra-3.1.90/plugins/karbonplugins/tools/CalligraphyTool/KarbonCalligraphyTool.h
--- calligra-3.1.90/plugins/karbonplugins/tools/CalligraphyTool/KarbonCalligraphyTool.h.omv~	2020-04-08 18:41:42.136084023 +0200
+++ calligra-3.1.90/plugins/karbonplugins/tools/CalligraphyTool/KarbonCalligraphyTool.h	2020-04-08 18:41:47.957215838 +0200
@@ -23,6 +23,7 @@
 #include <KoToolBase.h>
 #include <KoPathShape.h>
 #include <QPointer>
+#include <QPainterPath>
 
 class KoPathShape;
 class KarbonCalligraphicShape;
diff -up calligra-3.1.90/plugins/musicshape/MusicStyle.cpp.omv~ calligra-3.1.90/plugins/musicshape/MusicStyle.cpp
--- calligra-3.1.90/plugins/musicshape/MusicStyle.cpp.omv~	2020-04-08 18:00:34.232369082 +0200
+++ calligra-3.1.90/plugins/musicshape/MusicStyle.cpp	2020-04-08 18:01:08.860078353 +0200
@@ -17,6 +17,7 @@
  * Boston, MA 02110-1301, USA.
  */
 #include "MusicStyle.h"
+#include <QPainterPath>
 using namespace MusicCore;
 
 MusicStyle::MusicStyle()
diff -up calligra-3.1.90/plugins/musicshape/Renderer.cpp.omv~ calligra-3.1.90/plugins/musicshape/Renderer.cpp
--- calligra-3.1.90/plugins/musicshape/Renderer.cpp.omv~	2020-04-08 18:01:55.187063622 +0200
+++ calligra-3.1.90/plugins/musicshape/Renderer.cpp	2020-04-08 18:02:04.082257292 +0200
@@ -33,6 +33,7 @@
 #include "core/StaffSystem.h"
 
 #include <QMultiMap>
+#include <QPainterPath>
 
 #include <climits>
 #include <algorithm>
diff -up calligra-3.1.90/plugins/pathshapes/enhancedpath/EnhancedPathShape.cpp.omv~ calligra-3.1.90/plugins/pathshapes/enhancedpath/EnhancedPathShape.cpp
--- calligra-3.1.90/plugins/pathshapes/enhancedpath/EnhancedPathShape.cpp.omv~	2020-04-08 18:04:28.746585664 +0200
+++ calligra-3.1.90/plugins/pathshapes/enhancedpath/EnhancedPathShape.cpp	2020-04-08 18:04:36.550773835 +0200
@@ -36,6 +36,8 @@
 #include <KoOdfWorkaround.h>
 #include <KoPathPoint.h>
 
+#include <QPainterPath>
+
 EnhancedPathShape::EnhancedPathShape(const QRect &viewBox)
 : m_viewBox(viewBox)
 , m_viewBoxOffset(0.0, 0.0)
diff -up calligra-3.1.90/plugins/textshape/TextShape.cpp.omv~ calligra-3.1.90/plugins/textshape/TextShape.cpp
--- calligra-3.1.90/plugins/textshape/TextShape.cpp.omv~	2020-04-08 18:29:58.437444655 +0200
+++ calligra-3.1.90/plugins/textshape/TextShape.cpp	2020-04-08 18:30:05.427657721 +0200
@@ -57,6 +57,7 @@
 #include <QApplication>
 #include <QFont>
 #include <QPainter>
+#include <QPainterPath>
 #include <QPen>
 #include <QTextLayout>
 
diff -up calligra-3.1.90/sheets/ui/SheetView.cpp.omv~ calligra-3.1.90/sheets/ui/SheetView.cpp
--- calligra-3.1.90/sheets/ui/SheetView.cpp.omv~	2020-04-08 17:36:53.010307918 +0200
+++ calligra-3.1.90/sheets/ui/SheetView.cpp	2020-04-08 17:36:59.276249515 +0200
@@ -23,6 +23,7 @@
 #include <QCache>
 #include <QRect>
 #include <QPainter>
+#include <QPainterPath>
 #ifdef CALLIGRA_SHEETS_MT
 #include <QMutex>
 #include <QMutexLocker>
diff -up calligra-3.1.90/stage/part/KPrPresentationHighlightWidget.cpp.omv~ calligra-3.1.90/stage/part/KPrPresentationHighlightWidget.cpp
--- calligra-3.1.90/stage/part/KPrPresentationHighlightWidget.cpp.omv~	2020-04-08 16:56:52.789294808 +0200
+++ calligra-3.1.90/stage/part/KPrPresentationHighlightWidget.cpp	2020-04-08 16:56:56.430278571 +0200
@@ -26,6 +26,7 @@
 #include <QMouseEvent>
 #include <QColor>
 #include <QPainter>
+#include <QPainterPath>
 
 KPrPresentationHighlightWidget::KPrPresentationHighlightWidget( KoPACanvasBase * canvas )
 : KPrPresentationToolEventForwarder(canvas)
diff -up calligra-3.1.90/stage/plugins/pageeffects/edgewipe/barndoorwipe/BarnDoorWipeStrategy.cpp.omv~ calligra-3.1.90/stage/plugins/pageeffects/edgewipe/barndoorwipe/BarnDoorWipeStrategy.cpp
--- calligra-3.1.90/stage/plugins/pageeffects/edgewipe/barndoorwipe/BarnDoorWipeStrategy.cpp.omv~	2020-04-08 17:18:19.115901676 +0200
+++ calligra-3.1.90/stage/plugins/pageeffects/edgewipe/barndoorwipe/BarnDoorWipeStrategy.cpp	2020-04-08 17:18:22.778845426 +0200
@@ -21,6 +21,7 @@
 #include "BarnDoorWipeEffectFactory.h"
 #include <QWidget>
 #include <QPainter>
+#include <QPainterPath>
 
 const int StepCount = 250;
 
diff -up calligra-3.1.90/stage/plugins/pageeffects/edgewipe/barnveewipe/BarnVeeWipeStrategy.cpp.omv~ calligra-3.1.90/stage/plugins/pageeffects/edgewipe/barnveewipe/BarnVeeWipeStrategy.cpp
--- calligra-3.1.90/stage/plugins/pageeffects/edgewipe/barnveewipe/BarnVeeWipeStrategy.cpp.omv~	2020-04-08 17:21:39.837786610 +0200
+++ calligra-3.1.90/stage/plugins/pageeffects/edgewipe/barnveewipe/BarnVeeWipeStrategy.cpp	2020-04-08 17:21:43.254733072 +0200
@@ -21,6 +21,7 @@
 #include "BarnVeeWipeEffectFactory.h"
 #include <QWidget>
 #include <QPainter>
+#include <QPainterPath>
 
 const int StepCount = 250;
 
diff -up calligra-3.1.90/stage/plugins/pageeffects/edgewipe/barnzigzagwipe/BarnZigZagWipeStrategy.cpp.omv~ calligra-3.1.90/stage/plugins/pageeffects/edgewipe/barnzigzagwipe/BarnZigZagWipeStrategy.cpp
--- calligra-3.1.90/stage/plugins/pageeffects/edgewipe/barnzigzagwipe/BarnZigZagWipeStrategy.cpp.omv~	2020-04-08 17:24:56.131689971 +0200
+++ calligra-3.1.90/stage/plugins/pageeffects/edgewipe/barnzigzagwipe/BarnZigZagWipeStrategy.cpp	2020-04-08 17:24:59.540635859 +0200
@@ -21,6 +21,7 @@
 #include "BarnZigZagWipeEffectFactory.h"
 #include <QWidget>
 #include <QPainter>
+#include <QPainterPath>
 
 const int StepCount = 250;
 
diff -up calligra-3.1.90/stage/plugins/pageeffects/edgewipe/boxwipe/BoxWipeStrategy.cpp.omv~ calligra-3.1.90/stage/plugins/pageeffects/edgewipe/boxwipe/BoxWipeStrategy.cpp
--- calligra-3.1.90/stage/plugins/pageeffects/edgewipe/boxwipe/BoxWipeStrategy.cpp.omv~	2020-04-08 17:18:56.904320205 +0200
+++ calligra-3.1.90/stage/plugins/pageeffects/edgewipe/boxwipe/BoxWipeStrategy.cpp	2020-04-08 17:19:01.605247690 +0200
@@ -21,6 +21,7 @@
 #include "BoxWipeEffectFactory.h"
 #include <QWidget>
 #include <QPainter>
+#include <QPainterPath>
 
 const int StepCount = 250;
 
diff -up calligra-3.1.90/stage/plugins/pageeffects/edgewipe/diagonalwipe/DiagonalWipeStrategy.cpp.omv~ calligra-3.1.90/stage/plugins/pageeffects/edgewipe/diagonalwipe/DiagonalWipeStrategy.cpp
--- calligra-3.1.90/stage/plugins/pageeffects/edgewipe/diagonalwipe/DiagonalWipeStrategy.cpp.omv~	2020-04-08 17:12:26.803161811 +0200
+++ calligra-3.1.90/stage/plugins/pageeffects/edgewipe/diagonalwipe/DiagonalWipeStrategy.cpp	2020-04-08 17:12:30.085114532 +0200
@@ -21,6 +21,7 @@
 #include "DiagonalWipeEffectFactory.h"
 #include <QWidget>
 #include <QPainter>
+#include <QPainterPath>
 
 const int StepCount = 250;
 
diff -up calligra-3.1.90/stage/plugins/pageeffects/edgewipe/fourboxwipe/CornersInWipeStrategy.cpp.omv~ calligra-3.1.90/stage/plugins/pageeffects/edgewipe/fourboxwipe/CornersInWipeStrategy.cpp
--- calligra-3.1.90/stage/plugins/pageeffects/edgewipe/fourboxwipe/CornersInWipeStrategy.cpp.omv~	2020-04-08 17:03:20.734200622 +0200
+++ calligra-3.1.90/stage/plugins/pageeffects/edgewipe/fourboxwipe/CornersInWipeStrategy.cpp	2020-04-08 17:03:25.635147844 +0200
@@ -21,6 +21,7 @@
 #include "FourBoxWipeEffectFactory.h"
 #include <QWidget>
 #include <QPainter>
+#include <QPainterPath>
 
 const int StepCount = 250;
 
diff -up calligra-3.1.90/stage/plugins/pageeffects/edgewipe/fourboxwipe/CornersOutWipeStrategy.cpp.omv~ calligra-3.1.90/stage/plugins/pageeffects/edgewipe/fourboxwipe/CornersOutWipeStrategy.cpp
--- calligra-3.1.90/stage/plugins/pageeffects/edgewipe/fourboxwipe/CornersOutWipeStrategy.cpp.omv~	2020-04-08 17:04:07.662684864 +0200
+++ calligra-3.1.90/stage/plugins/pageeffects/edgewipe/fourboxwipe/CornersOutWipeStrategy.cpp	2020-04-08 17:04:11.191645171 +0200
@@ -21,6 +21,7 @@
 #include "FourBoxWipeEffectFactory.h"
 #include <QWidget>
 #include <QPainter>
+#include <QPainterPath>
 
 const int StepCount = 250;
 
diff -up calligra-3.1.90/stage/plugins/pageeffects/edgewipe/miscdiagonalwipe/DoubleBarnDoorWipeStrategy.cpp.omv~ calligra-3.1.90/stage/plugins/pageeffects/edgewipe/miscdiagonalwipe/DoubleBarnDoorWipeStrategy.cpp
--- calligra-3.1.90/stage/plugins/pageeffects/edgewipe/miscdiagonalwipe/DoubleBarnDoorWipeStrategy.cpp.omv~	2020-04-08 17:01:45.315172810 +0200
+++ calligra-3.1.90/stage/plugins/pageeffects/edgewipe/miscdiagonalwipe/DoubleBarnDoorWipeStrategy.cpp	2020-04-08 17:01:49.780129818 +0200
@@ -21,6 +21,7 @@
 #include "MiscDiagonalWipeEffectFactory.h"
 #include <QWidget>
 #include <QPainter>
+#include <QPainterPath>
 
 const int StepCount = 250;
 
diff -up calligra-3.1.90/stage/plugins/pageeffects/edgewipe/miscdiagonalwipe/DoubleDiamondWipeStrategy.cpp.omv~ calligra-3.1.90/stage/plugins/pageeffects/edgewipe/miscdiagonalwipe/DoubleDiamondWipeStrategy.cpp
--- calligra-3.1.90/stage/plugins/pageeffects/edgewipe/miscdiagonalwipe/DoubleDiamondWipeStrategy.cpp.omv~	2020-04-08 17:02:15.773874451 +0200
+++ calligra-3.1.90/stage/plugins/pageeffects/edgewipe/miscdiagonalwipe/DoubleDiamondWipeStrategy.cpp	2020-04-08 17:02:19.555836587 +0200
@@ -21,6 +21,7 @@
 #include "MiscDiagonalWipeEffectFactory.h"
 #include <QWidget>
 #include <QPainter>
+#include <QPainterPath>
 
 const int StepCount = 250;
 
diff -up calligra-3.1.90/stage/plugins/pageeffects/edgewipe/veewipe/VeeWipeStrategy.cpp.omv~ calligra-3.1.90/stage/plugins/pageeffects/edgewipe/veewipe/VeeWipeStrategy.cpp
--- calligra-3.1.90/stage/plugins/pageeffects/edgewipe/veewipe/VeeWipeStrategy.cpp.omv~	2020-04-08 17:20:24.690960047 +0200
+++ calligra-3.1.90/stage/plugins/pageeffects/edgewipe/veewipe/VeeWipeStrategy.cpp	2020-04-08 17:20:29.660882684 +0200
@@ -21,6 +21,7 @@
 #include "VeeWipeEffectFactory.h"
 #include <QWidget>
 #include <QPainter>
+#include <QPainterPath>
 
 const int StepCount = 250;
 
diff -up calligra-3.1.90/stage/plugins/pageeffects/edgewipe/zigzagwipe/ZigZagWipeStrategy.cpp.omv~ calligra-3.1.90/stage/plugins/pageeffects/edgewipe/zigzagwipe/ZigZagWipeStrategy.cpp
--- calligra-3.1.90/stage/plugins/pageeffects/edgewipe/zigzagwipe/ZigZagWipeStrategy.cpp.omv~	2020-04-08 17:22:45.972747920 +0200
+++ calligra-3.1.90/stage/plugins/pageeffects/edgewipe/zigzagwipe/ZigZagWipeStrategy.cpp	2020-04-08 17:22:48.840702761 +0200
@@ -21,6 +21,7 @@
 #include "ZigZagWipeEffectFactory.h"
 #include <QWidget>
 #include <QPainter>
+#include <QPainterPath>
 
 const int StepCount = 250;
 
diff -up calligra-3.1.90/words/part/frames/KWFrameLayout.cpp.omv~ calligra-3.1.90/words/part/frames/KWFrameLayout.cpp
--- calligra-3.1.90/words/part/frames/KWFrameLayout.cpp.omv~	2020-04-08 16:46:40.092215184 +0200
+++ calligra-3.1.90/words/part/frames/KWFrameLayout.cpp	2020-04-08 16:46:50.105442659 +0200
@@ -40,6 +40,7 @@
 #include <QTextLayout>
 #include <QTextDocument>
 #include <QTextBlock>
+#include <QPainterPath>
 #include <WordsDebug.h>
 #include <limits.h>
 
